lti_consumer/lti13.py

```diff
@@ -0,0 +1,187 @@
+
+    # Optional claims - Useful
+    # This is useful for error redirects
+    # http://www.imsglobal.org/spec/lti/v1p3/#launch-presentation-claim
+    "https://purl.imsglobal.org/spec/lti/claim/launch_presentation": {
+        "document_target": "iframe",  # iframe, frame, window
+        # Endpoint to redirect the user to after completing LTI task
+        # Returns with log statements
+        "return_url": "https://platform.example.edu/terms/201601/courses/7/sections/1/resources/2"
+    },
+    # Custom variables :)
+    "https://purl.imsglobal.org/spec/lti/claim/custom": {
+        "xstart": "2017-04-21T01:00:00Z",
+        "request_url": "https://tool.com/link/123"
+    }
+}
+
+
+class LtiConsumer1p3:
+    def __init__(
+            self,
+            iss,
+            lti_oidc_url,
+            lti_launch_url,
+            deployment_id,
+            rsa_key,
+    ):
+        self.iss = iss
+        self.oidc_url = lti_oidc_url
+        self.launch_url = lti_launch_url
+        self.deployment_id = deployment_id
+
+        # Generate JWK from RSA key
+        self.jwk = RSAKey(
+            # Don't hardcode key name
+            kid="lti_key",
+            key=RSA.import_key(rsa_key)
+        )
+
+    def _encode_and_sign(self, message):
+        # Dump JSON and encode it with key
+        msg = json.dumps(message)
+
+        # The class instance that sets up the signing operation
+        # An RS 256 key is required for LTI 1.3
+        _jws = JWS(msg, alg="RS256")
+
+        # Encode and sign LTI message
+        return _jws.sign_compact([self.jwk])
+
+    @staticmethod
+    def _get_user_roles(roles):
+        """
+        Converts platform roles into LTI compliant roles
+
+        Used in roles claim: should return array of URI values
+        for roles that the user has within the message's context.
+
+        Reference: http://www.imsglobal.org/spec/lti/v1p3/#roles-claim
+        Role vocabularies: http://www.imsglobal.org/spec/lti/v1p3/#role-vocabularies
+        """
+        return [
+            "http://purl.imsglobal.org/vocab/lis/v2/institution/person#Student",
+            "http://purl.imsglobal.org/vocab/lis/v2/membership#Learner",
+            "http://purl.imsglobal.org/vocab/lis/v2/membership#Mentor"
+        ]
+
+    def preprare_preflight_request(
+        self,
+        callback_url,
+        hint="oidc_hint",
+        lti_hint="lti_hint"
+    ):
+        """
+        Generates OIDC url with parameters
+        """
+        oidc_url = self.oidc_url + "?"
+        parameters = {
+            "iss": self.iss,
+            "target_link_uri": callback_url,
+            "login_hint": hint,
+            "lti_message_hint": lti_hint
+        }
+
+        return {
+            "oidc_url": oidc_url + urlencode(parameters),
+        }
+
+    def generate_launch_request(
+        self,
+        user_id,
+        roles,
+        resource_link,
+        preflight_response,
+    ):
+        data = {
+            **LTI_BASE_MESSAGE,
```

Comment:
Getting a SyntaxError trying to run this in `ironwood.master` due to Python2.7
```
edx.devstack.studio  |   File "/edx/src/xblock-lti-consumer/lti_consumer/lti13.py", line 133
edx.devstack.studio  |     **LTI_BASE_MESSAGE,
edx.devstack.studio  |      ^
edx.devstack.studio  | SyntaxError: invalid syntax
```

Reply:
Fixed!